### ISSUE TITLE
ppc64le throwing errors on arm64 and linux x86_64 when running go install

### ISSUE THREAD

**Issue by vanpelt** [go-delve/delve#3558]
Please answer the following before submitting your issue:

1. What version of Delve are you using (`dlv version`)? - The issue is present in 1.21.1 and 1.21.2.
3. What version of Go are you using? (`go version`)? - 1.21.3
4. What operating system and processor architecture are you using? - The error occurs on darwin/arm64 and linux/x86_64
5. What did you do? - `go install github.com/go-delve/delve/cmd/dlv`
6. What did you expect to see? - Successful install
7. What did you see instead? - The error below

```
# github.com/go-delve/delve/pkg/proc
../../../../go/pkg/mod/github.com/go-delve/delve@v1.21.1/pkg/proc/ppc64le_disasm.go:57:63: undefined: ppc64asm.RFI
../../../../go/pkg/mod/github.com/go-delve/delve@v1.21.1/pkg/proc/ppc64le_disasm.go:57:77: undefined: ppc64asm.RFCI
../../../../go/pkg/mod/github.com/go-delve/delve@v1.21.1/pkg/proc/ppc64le_disasm.go:57:92: undefined: ppc64asm.RFDI
../../../../go/pkg/mod/github.com/go-delve/delve@v1.21.1/pkg/proc/ppc64le_disasm.go:57:107: undefined: ppc64asm.RFMCI
../../../../go/pkg/mod/github.com/go-delve/delve@v1.21.1/pkg/proc/ppc64le_disasm.go:57:123: undefined: ppc64asm.RFGI
```

**Comment by vanpelt** [comment]
This is likely due to a breaking change in `golang.org/x/arch`.  I have `v0.6.0` in my go.mod.

**Comment by deepakgupta-remitly** [comment]
I am facing the same issue.
in delve versions 1.21.1 and 1.21.2

**Comment by derekparker** [comment]
I'm not able to reproduce locally. Are you running these commands from within the repository? No currently released version includes these dependency updates.

Can you try running `go clean -modcache`?

**Comment by derekparker** [comment]
> This is likely due to a breaking change in `golang.org/x/arch`. I have `v0.6.0` in my go.mod.

The current `go.mod` in master has `golang.org/x/arch@v0.5.0`.

**Comment by aarzilli** [comment]
I don't think we should be doing anything, eventually we'll update that dependency but we can't be expected to compile with dependency versions other than the ones listed.

**Comment by vanpelt** [comment]
I'm not sure what's going on here, maybe the `golang.org/x/arch` theory is flawed.  I was searching for the `RFDI` const in the [arch source](https://cs.opensource.google/go/x/arch/+/refs/tags/v0.6.0:ppc64/ppc64asm/tables.go) and not seeing it anywhere.

I just attempted to create a new go module and see if I could reproduce but I also wasn't able to.  When I ran `go get github.com/go-delve/delve/cmd/dlv` in this module it actually brought in `golang.org/x/arch v0.6.0` as an indirect dep.  I'll keep trying to get into a state where this error is occuring, @deepakgupta-remitly if you're able to create a minimal `go.mod` file that get's you into this state that would be really helpful for root causing.

**Comment by aarzilli** [comment]
A few constants where removed from golang.org/x/arch/ppc64/ppc64asm in some version of golang.org/x/arch. This problem is the result of trying to use some version of delve with a version of x/arch that is too recent. Delve itself has already been updated to work with newer versions of x/arch (but this isn't part of any released version yet).

Nothing to do here, now or in the future.
